lib/pages/BeginRidePage.dart

```diff
@@ -0,0 +1,178 @@
+
+class _StopCircle extends StatelessWidget {
+  final bool _dropoff;
+
+  _StopCircle(this._dropoff);
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(alignment: Alignment.center, children: [
+      Container(
+          width: 26,
+          height: 26,
+          decoration: new BoxDecoration(
+            color: Colors.black,
+            shape: BoxShape.circle,
+          )),
+      Container(
+          width: 8,
+          height: 8,
+          decoration: new BoxDecoration(
+            color: Colors.white,
+            shape: _dropoff ? BoxShape.rectangle : BoxShape.circle,
+          ))
+    ]);
+  }
+}
+
+class _StopData {
+  final bool dropoff;
+  final DateTime time;
+  final String stop;
+  final String address;
+
+  _StopData(this.dropoff, this.time, this.stop, this.address);
+}
+
+class Stops extends StatefulWidget {
+  final List<_StopData> stops;
+
+  const Stops({Key key, @required this.stops}) : super(key: key);
+
+  @override
+  StopsState createState() {
+    return StopsState();
+  }
+}
+
+class StopsState extends State<Stops> {
+  double _height = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Expanded(
+      child: Stack(children: [
+        Container(
+          width: 52,
+          height: _height,
+          alignment: Alignment.topCenter,
+          child: Container(
+            width: 4,
+            decoration: new BoxDecoration(
+                color: const Color(0xFFECEBED),
+                shape: BoxShape.rectangle,
+                borderRadius: BorderRadius.all(Radius.circular(8.0))),
+          ),
+        ),
+        MeasureSize(
+          onChange: (size) {
+            setState(() {
+              _height = size.height;
+            });
+          },
+          child: ListView.separated(
+              shrinkWrap: true,
+              itemCount: widget.stops.length,
+              separatorBuilder: (context, index) {
+                return SizedBox(height: 32);
+              },
+              itemBuilder: (context, index) {
+                final stop = widget.stops[index];
+                return Padding(
+                  padding: const EdgeInsets.only(right: 10),
+                  child: Row(
+                    children: [
+                      Container(width: 52, child: _StopCircle(stop.dropoff)),
+                      Expanded(
+                        child: RideDestPickupCard(
+                            stop.dropoff, stop.time, stop.stop, stop.address),
+                      ),
+                    ],
+                  ),
+                );
+              }),
+        ),
+      ]),
+    );
+  }
+}
+
+// TODO: replace with real model later
+class _TempPageData {
+  final String firstName;
+  final ImageProvider<dynamic> photo;
+  final DateTime time;
+  final String stop;
+  final String address;
+
+  final List<_StopData> stops;
+
+  _TempPageData(this.firstName, this.photo, this.time, this.stop, this.address,
+      this.stops);
+}
```

Comment:
Even though this is very similar to the other class you defined, it include stops as well, so this one can stay.

Reply:
I replaced it to use the same one for all three if that's okay